lib/internal/eachOfLimit.js

```diff
@@ -14,17 +14,23 @@ export default function _eachOfLimit(limit) {
         }
         var nextElem = iterator(obj);
         var done = false;
+        var canceled = false;
         var running = 0;
         var looping = false;
 
         function iterateeCallback(err, value) {
             running -= 1;
-            if (err) {
+            if (err && !canceled) {
                 done = true;
                 callback(err);
             }
+            else if (err === false) {
+                done = true;
+                canceled = true;
+            }
             else if (value === breakLoop || (done && running <= 0)) {
                 done = true;
+                if (canceled) return
```

Comment:
Move to top of `iterateeCallback` for simplicity? (it would also let you avoid the `!canceled` check in the error condition)